apps/backend/src/datasources/TechniqueDataSource.ts

```diff
@@ -37,5 +37,8 @@ export interface TechniqueDataSource {
     techniqueId: number
   ): Promise<boolean>;
   getTechniquesByProposalPk(proposalPk: number): Promise<Technique[]>;
-  checkProposalsHasTechniques(proposalPks: number[]): Promise<boolean>;
+  isXpressInstrumentAndProposal(
```

Comment:
Might be better to split the Xpress check to the STFC-specific mutation and keep the datasource to checking if the proposal belongs to the technique. It's a bit less optimised but then keeps the datasource generic and free of Xpress logic. Happy if you think it's best left though